pkg/process/process.go

```diff
@@ -257,6 +257,9 @@ func AddCloneEvent(event *tetragonAPI.MsgCloneEvent) error {
 		pi.process.ParentExecId = parentExecId
 		pi.process.ExecId = GetProcessID(event.PID, event.Ktime)
 		pi.process.Pid = &wrapperspb.UInt32Value{Value: event.PID}
+		// Add TID of the thread leader so it will be cached. Then on non execve events
+		// the TID of the triggered events will be fetched on the fly since we do not track it.
+		pi.process.Tid = &wrapperspb.UInt32Value{Value: event.TID}
```

Comment:
I don't have a strong opinion on adding `tid` into the process cache or not. We have to override that whenever we are going to generate any user events. Based on that we don't need to set this here, but it also does not cause any issues. 

Maybe for debugging purposes, we can set this here to an invalid value (i.e. `UINT32_MAX`?) and we should also check that we don't generate any events with that (wrong) value.

Reply:
> I don't have a strong opinion on adding `tid` into the process cache or not. We have to override that whenever we are going to generate any user events. Based on that we don't need to set this here, but it also does not cause any issues.
> 
> Maybe for debugging purposes, we can set this here to an invalid value (i.e. `UINT32_MAX`?) and we should also check that we don't generate any events with that (wrong) value.

That makes sense indeed, @kkourt had same suggestion to not put into the cache but have that value 0 , then later we can use it to know that we didn't add the execve event as an indication! so will use 0 as it is an invalid pid, and see how to add a final log message at the execve to catch/assert on this eventually!

For the clone event sending TID, we had a long discussion and I don't think it hurts having it, especially since I added BPF tests and debug messages to assert on it, and the updated version is clear about what we expect. So I guess I will keep it! thank you @tpapagian !